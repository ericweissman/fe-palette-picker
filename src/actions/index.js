export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
});

//projects
export const getProjectsSuccess = (projects) => ({
  type: 'GET_PROJECTS_SUCCESS',
  projects
})

export const addProjectSuccess = (project) => ({
  type: 'ADD_PROJECT_SUCCESS',
  project
})