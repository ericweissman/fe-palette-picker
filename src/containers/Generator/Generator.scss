.generator {
  h1 {
    font-family: $title-font;
    font-size: 2.75rem;
    text-align: center;
    margin: 10px;
  }
  display: flex;
  flex-direction: column;
  align-items: center;
}

.palette-main {
  display: flex;
}