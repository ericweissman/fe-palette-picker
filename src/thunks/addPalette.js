import { isLoading, hasErrored, addPaletteSuccess } from '../actions'

export const addPalette = (url, palette) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "POST",
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(palette)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(addPaletteSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}