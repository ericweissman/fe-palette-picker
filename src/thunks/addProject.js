import { isLoading, hasErrored, addProjectSuccess } from '../actions'

export const addProject = (url, project_name) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "POST",
        headers: {
          "Content-Type" : "application/json"
        },
        body: JSON.stringify({ project_name })
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(addProjectSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}