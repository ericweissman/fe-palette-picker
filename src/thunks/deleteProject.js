import { isLoading, hasErrored, deleteProjectSuccess } from '../actions'

export const deleteProject = (url, projectID) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "DELETE",
        headers: {
          "Content-Type" : "application/json"
        }
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      // const result = await response.json()
      dispatch(deleteProjectSuccess(projectID))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}