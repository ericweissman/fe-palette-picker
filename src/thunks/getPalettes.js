import { isLoading, hasErrored, getPalettesSuccess } from '../actions'

export const getPalettes = (url, id) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(getPalettesSuccess(result, id))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}