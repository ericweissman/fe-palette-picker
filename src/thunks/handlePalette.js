import { isLoading, hasErrored } from '../actions'

export const handlePalette = (url, actionToDispatch, method, palette) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: `${method}`,
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(palette)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(actionToDispatch(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}