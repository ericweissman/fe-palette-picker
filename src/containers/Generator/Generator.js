import React, { Component } from 'react'
import { connect } from 'react-redux';
import { setActivePalette, editPaletteSuccess } from '../../actions';
import { handlePalette } from '../../thunks/handlePalette';
import PropTypes from 'prop-types';
import Swatch from '../../components/Swatch/Swatch'
import '../../main.scss';

export class Generator extends Component {
  constructor() {
    super()
    this.state = {
      locked: [],
    }
  }

  componentDidMount() {
    if (!this.props.activePalette.length) {
      this.generatePalette()
    }
  }

  createColor = () => {
    const options = '0123456789ABCDEF'
    let hexArray = ['#']
    for (let i = 0; i < 6; i++) {
      let random = Math.floor(Math.random() * (16))
      hexArray.push(options[random])
    }
    return hexArray.join('')
  }

  generatePalette = () => {
    const { locked } = this.state
    const { activePalette } = this.props
    let palette = []
    for (let i = 0; i < 5; i++) {
      let color = ''
      if (locked.includes(i)) {
        color = activePalette[i]
      } else {
        color = this.createColor()
      }
      palette.push(color)
    }
    this.props.setActivePalette(palette)
  }

  handleClick = (e) => {
    const { value } = e.target
    const { locked } = this.state
    let checked = []
    if (locked.includes(parseInt(value))) {
      checked = locked.filter(id => id !== parseInt(value))
    } else {
      checked = [...locked, parseInt(value)]
    }
    this.setState({
      locked: checked,
    })
  }

  savePalette = () => {
    const { id, project_id, palette_name} = this.props.paletteToEdit
    const { activePalette} = this.props
    const palette = {
      palette_name: palette_name,
      color_1: activePalette[0],
      color_2: activePalette[1],
      color_3: activePalette[2],
      color_4: activePalette[3],
      color_5: activePalette[4]
    }
    const url = process.env.REACT_APP_BACKEND_URL + `/api/v1/projects/${project_id}/palettes/${id}` 
    this.props.handlePalette(url, editPaletteSuccess, 'PUT', palette)
  }


  render() {
    const { activePalette } = this.props
    return (
      <div className='generator'>
        <h1 className="title">Palette Picker</h1>
        <div className='palette-main'>
          {
            activePalette.map((palette, i) => {
              return (
                <Swatch key={i} value={i} palette={palette} handleClick={this.handleClick} />
              )
            })
          }
        </div>
        <div className='generator-btns'>
          <button onClick={this.generatePalette}>Generate Palette</button>
          <button onClick={this.savePalette}>Save Palette</button>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  activePalette: state.activePalette,
  paletteToEdit: state.paletteToEdit,
})

export const mapDispatchToProps = (dispatch) => ({
  setActivePalette: (palette) => dispatch(setActivePalette(palette)),
  handlePalette: (url, actionToDispatch, method, palette) => dispatch(handlePalette(url, actionToDispatch, method, palette)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Generator);

Generator.propTypes = {
  handlePalette: PropTypes.func,
  setActivePalette: PropTypes.func,
  activePalette: PropTypes.array,
  paletteToEdit: PropTypes.object,
}