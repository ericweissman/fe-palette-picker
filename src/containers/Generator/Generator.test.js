import React from 'react';
import { shallow } from 'enzyme';
import { handlePalette } from '../../thunks/handlePalette'
import { Generator, mapStateToProps, mapDispatchToProps } from './Generator'
import { setActivePalette } from '../../actions'

jest.mock('../../thunks/handlePalette')

describe('Generator', () => {
  let wrapper

  describe('snapshots and initial state', () => {

    it('should have an initial state', () => {
      const mockPaletteToEdit = {}
      const mockActivePalette = []
      const setActivePalette = jest.fn()

      const expected = { locked: [] }
      wrapper = shallow(
        <Generator paletteToEdit={mockPaletteToEdit} activePalette={mockActivePalette} setActivePalette={setActivePalette}  />
      )

      expect(wrapper.state()).toEqual(expected)
    })

    it('should match the initial snapshot', () => {
      const mockPaletteToEdit = {}
      const mockActivePalette = ['default']
      const setActivePalette = jest.fn()

      wrapper = shallow(
        <Generator paletteToEdit={mockPaletteToEdit} activePalette={mockActivePalette} setActivePalette={setActivePalette} />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with paletteToEdit and activePalette', () => {
      const mockPaletteToEdit = {
        palette_name: "new",
        id: 21,
        project_id: 10
      }
      const mockActivePalette = ['111111', '222222', '333333', '444444', '555555']

      wrapper = shallow(
        <Generator paletteToEdit={mockPaletteToEdit} activePalette={mockActivePalette} />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with a paletteToEdit object and activePalette array', () => {
      const mockState = { paletteToEdit: { palette_name: "new", id: 21, project_id: 10 }, activePalette: ['1', '2', '3', '4', '5'], palettes: [], projects: [] }
      const expected = { paletteToEdit: { palette_name: "new", id: 21, project_id: 10 }, activePalette: ['1', '2', '3', '4', '5'] }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call handlePalette dispatch', () => {
      const mockDispatch = jest.fn()
      const mockURL = 'http://localhost:3001/api/v1/projects/1/palettes'
      const mockPalette = {}

      const actionToDispatch = handlePalette(mockURL, setActivePalette, 'POST', mockPalette)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.handlePalette(mockURL, setActivePalette, 'POST', mockPalette)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call setActivePalette dispatch', () => {
      const mockDispatch = jest.fn()
      const mockPalette = ['1', '2', '3', '4', '5']

      const actionToDispatch = setActivePalette(mockPalette)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setActivePalette(mockPalette)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})