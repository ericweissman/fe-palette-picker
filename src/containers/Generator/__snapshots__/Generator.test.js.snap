// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator snapshots and initial state should match the initial snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Generator
    activePalette={
      Array [
        "default",
      ]
    }
    paletteToEdit={Object {}}
    setActivePalette={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Palette Picker
        </h1>,
        <div
          className="palette-main"
        >
          <Swatch
            handleClick={[Function]}
            palette="default"
            value={0}
          />
        </div>,
        <div
          className="generator-btns"
        >
          <button
            onClick={[Function]}
          >
            Generate Palette
          </button>
          <button
            onClick={[Function]}
          >
            Save Palette
          </button>
        </div>,
      ],
      "className": "generator",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Palette Picker",
        },
        "ref": null,
        "rendered": "Palette Picker",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Swatch
              handleClick={[Function]}
              palette="default"
              value={0}
            />,
          ],
          "className": "palette-main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "handleClick": [Function],
              "palette": "default",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
            >
              Generate Palette
            </button>,
            <button
              onClick={[Function]}
            >
              Save Palette
            </button>,
          ],
          "className": "generator-btns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Generate Palette",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Generate Palette",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save Palette",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Save Palette",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Palette Picker
          </h1>,
          <div
            className="palette-main"
          >
            <Swatch
              handleClick={[Function]}
              palette="default"
              value={0}
            />
          </div>,
          <div
            className="generator-btns"
          >
            <button
              onClick={[Function]}
            >
              Generate Palette
            </button>
            <button
              onClick={[Function]}
            >
              Save Palette
            </button>
          </div>,
        ],
        "className": "generator",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Palette Picker",
          },
          "ref": null,
          "rendered": "Palette Picker",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Swatch
                handleClick={[Function]}
                palette="default"
                value={0}
              />,
            ],
            "className": "palette-main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "handleClick": [Function],
                "palette": "default",
                "value": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
              >
                Generate Palette
              </button>,
              <button
                onClick={[Function]}
              >
                Save Palette
              </button>,
            ],
            "className": "generator-btns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Generate Palette",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Generate Palette",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save Palette",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Save Palette",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Generator snapshots and initial state should match the snapshot with paletteToEdit and activePalette 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Generator
    activePalette={
      Array [
        "111111",
        "222222",
        "333333",
        "444444",
        "555555",
      ]
    }
    paletteToEdit={
      Object {
        "id": 21,
        "palette_name": "new",
        "project_id": 10,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Palette Picker
        </h1>,
        <div
          className="palette-main"
        >
          <Swatch
            handleClick={[Function]}
            palette="111111"
            value={0}
          />
          <Swatch
            handleClick={[Function]}
            palette="222222"
            value={1}
          />
          <Swatch
            handleClick={[Function]}
            palette="333333"
            value={2}
          />
          <Swatch
            handleClick={[Function]}
            palette="444444"
            value={3}
          />
          <Swatch
            handleClick={[Function]}
            palette="555555"
            value={4}
          />
        </div>,
        <div
          className="generator-btns"
        >
          <button
            onClick={[Function]}
          >
            Generate Palette
          </button>
          <button
            onClick={[Function]}
          >
            Save Palette
          </button>
        </div>,
      ],
      "className": "generator",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Palette Picker",
        },
        "ref": null,
        "rendered": "Palette Picker",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Swatch
              handleClick={[Function]}
              palette="111111"
              value={0}
            />,
            <Swatch
              handleClick={[Function]}
              palette="222222"
              value={1}
            />,
            <Swatch
              handleClick={[Function]}
              palette="333333"
              value={2}
            />,
            <Swatch
              handleClick={[Function]}
              palette="444444"
              value={3}
            />,
            <Swatch
              handleClick={[Function]}
              palette="555555"
              value={4}
            />,
          ],
          "className": "palette-main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "handleClick": [Function],
              "palette": "111111",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "handleClick": [Function],
              "palette": "222222",
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "handleClick": [Function],
              "palette": "333333",
              "value": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "handleClick": [Function],
              "palette": "444444",
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "class",
            "props": Object {
              "handleClick": [Function],
              "palette": "555555",
              "value": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
            >
              Generate Palette
            </button>,
            <button
              onClick={[Function]}
            >
              Save Palette
            </button>,
          ],
          "className": "generator-btns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Generate Palette",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Generate Palette",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save Palette",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Save Palette",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Palette Picker
          </h1>,
          <div
            className="palette-main"
          >
            <Swatch
              handleClick={[Function]}
              palette="111111"
              value={0}
            />
            <Swatch
              handleClick={[Function]}
              palette="222222"
              value={1}
            />
            <Swatch
              handleClick={[Function]}
              palette="333333"
              value={2}
            />
            <Swatch
              handleClick={[Function]}
              palette="444444"
              value={3}
            />
            <Swatch
              handleClick={[Function]}
              palette="555555"
              value={4}
            />
          </div>,
          <div
            className="generator-btns"
          >
            <button
              onClick={[Function]}
            >
              Generate Palette
            </button>
            <button
              onClick={[Function]}
            >
              Save Palette
            </button>
          </div>,
        ],
        "className": "generator",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Palette Picker",
          },
          "ref": null,
          "rendered": "Palette Picker",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Swatch
                handleClick={[Function]}
                palette="111111"
                value={0}
              />,
              <Swatch
                handleClick={[Function]}
                palette="222222"
                value={1}
              />,
              <Swatch
                handleClick={[Function]}
                palette="333333"
                value={2}
              />,
              <Swatch
                handleClick={[Function]}
                palette="444444"
                value={3}
              />,
              <Swatch
                handleClick={[Function]}
                palette="555555"
                value={4}
              />,
            ],
            "className": "palette-main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "handleClick": [Function],
                "palette": "111111",
                "value": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "handleClick": [Function],
                "palette": "222222",
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "handleClick": [Function],
                "palette": "333333",
                "value": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "handleClick": [Function],
                "palette": "444444",
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "class",
              "props": Object {
                "handleClick": [Function],
                "palette": "555555",
                "value": 4,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
              >
                Generate Palette
              </button>,
              <button
                onClick={[Function]}
              >
                Save Palette
              </button>,
            ],
            "className": "generator-btns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Generate Palette",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Generate Palette",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save Palette",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Save Palette",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
